src/Common/src/TypeSystem/Common/InstantiatedType.cs

```diff
@@ -281,5 +281,21 @@ public override bool HasCustomAttribute(string attributeNamespace, string attrib
         {
             return _typeDef.HasCustomAttribute(attributeNamespace, attributeName);
         }
+
+        public override MetadataType GetNestedType(string name)
+        {
+            // It doesn't make sense to look at the types nested in an instantiated type in general.
+            // In particular, if the outer type is generic, and the inner type has an unrelated set of 
+            // generic parameters, the result is certain to be confusing.
+            throw new NotImplementedException();
```

Comment:
Well, that could also be done. I prefer the throwing approach as returning the open nested types is known to be confusing to consumer's of the api, but I'd like to see what Michal's opinion is.